From 23bed0400b70bbc7e8eed9a5a3a912bcbe645c7d Mon Sep 17 00:00:00 2001
From: Anton Bambura <jenneron@protonmail.com>
Date: Tue, 19 Jan 2021 11:22:31 +0200
Subject: [PATCH 169/210] arm64: dts: qcom: msm8916-wingtech-wt86528: add
 sensors

This device has MPU-6880 imu and Asahi Kasei AK09911 magnetometer. Add
support for it.

Signed-off-by: Anton Bambura <jenneron@protonmail.com>
---
 .../dts/qcom/msm8916-wingtech-wt86528.dts     | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts
index a77f8e2e408..7cc043db75d 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts
@@ -38,6 +38,34 @@ usb_id: usb-id {
 	};
 };
 
+&blsp_i2c2 {
+	magnetometer@c {
+		compatible = "asahi-kasei,ak09911";
+		reg = <0x0c>;
+
+		vdd-supply = <&pm8916_l17>;
+		vid-supply = <&pm8916_l6>;
+	};
+
+	imu@68 {
+		compatible = "invensense,mpu6880";
+		reg = <0x68>;
+
+		interrupt-parent = <&msmgpio>;
+		interrupts = <115 IRQ_TYPE_EDGE_RISING>;
+
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&imu_default>;
+
+		mount-matrix = "1",  "0", "0",
+			       "0", "-1", "0",
+			       "0",  "0", "1";
+	};
+};
+
 &reg_lcd_neg {
 	regulator-min-microvolt = <5000000>;
 	regulator-max-microvolt = <5000000>;
@@ -65,6 +93,14 @@ gpio_leds_default: gpio-leds-default {
 		bias-disable;
 	};
 
+	imu_default: imu-default {
+		pins = "gpio115";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
 	usb_id_default: usb-id-default {
 		pins = "gpio110";
 		function = "gpio";
-- 
2.34.1

